Initial implementation on iNZight

szmsu2011 opened the pull request.

szmsu2011 committed:
setup changes

szmsu2011 committed:
multi-key gui

tmelliott commented:
Things I've noticed:
- [x] default subset focus is "level 1", rather than "none" (/"all")
- [x] the 'key' multiselect is small (this may have been/probably was present before you started)
- [x] lots of levels = slow, and plots are mostly spaghetti
- [x] base-R is best where possible (reduces the number of things that will break whenever dplyr/purrr change)
e.g., 
```r
kv <- ts_object |>
    tsibble::key_data() |>
    dplyr::select(-.rows) |>
    purrr::array_branch(1) |>
    purrr::map_chr(\(x) paste(x, collapse = "/"))
```
vs
```r
kv <- ts_object |>
    attr("key") |> 
    apply(1, \(x) paste(x[1:(length(x) - 1)], collapse = "/"))
```

szmsu2011 committed:
optimise workflow

tmelliott committed:
Multi-vars bug, hide slider

Fix small bug in `measure_var` check (could be >1 value); also change some enabled() to visible() for the g_subset group.

tmelliott committed:
Merge pull request #2 from iNZightVIT/tmelliott/edits

Multi-vars bug, hide slider - @szmsu2011 I'll let you merge this (if you want to)

szmsu2011 committed:
new ui control logic

tmelliott commented:
For sliders, I generally use timers to reduce lag. E.g., from https://github.com/iNZightVIT/iNZight/blob/dev/R/iNZPlotModWin.R#LL2985C13-L2999C14
```r
addHandlerChanged(bgCol,
    handler = function(h, ...) {
        if (!is.null(timer)) {
            if (timer$started) timer$stop_timer()
        }
        timer <<- gtimer(500,
            function(...) {
                if (nchar(svalue(bgCol)) >= 3) {
                    updateEverything()
                }
            },
            one.shot = TRUE
        )
    }
)
```

tmelliott commented:
- [x] focus subset not implemented for forecasts
- [x] smoother visible for categorical plot

tmelliott committed:
update desc

szmsu2011 committed:
ui fixes and optimisations

szmsu2011 committed:
add range filters

tmelliott committed:
fix: Change sliders to dropdowns

tmelliott committed:
remove superfluous code

szmsu2011 committed:
toggle for mult_fit and update range filters

szmsu2011 committed:
Merge pull request #3 from iNZightVIT/tmelliott/change-slider-to-dropdown

fix: Change sliders to dropdowns
